typescript/api/services/RecordsService.ts

```diff
@@ -756,9 +757,37 @@ export module Services {
       return await this.storageService.createRecordAudit(record);
     }
 
+    public async transitionWorkflowStep(currentRec: any, recordType: any, nextStep: any, user: any, triggerPreSaveTriggers: boolean = true, triggerPostSaveTriggers: boolean = true) {
 
-    public updateWorkflowStep(currentRec, nextStep): void {
+      let oid = _.get(currentRec, 'redboxOid');
+      let updateResponse = new StorageServiceResponse();
+      updateResponse.oid = oid;
+      updateResponse.success = true;
+      
       if (!_.isEmpty(nextStep)) {
+        if (triggerPreSaveTriggers) {
+          currentRec = await this.triggerPreSaveTriggers(oid, currentRec, recordType, 'onTransitionWorkflow', user);
+        }
+
+        this.setWorkflowStepInMetaMetadata(currentRec,nextStep);
+
+        if (triggerPostSaveTriggers) {
+
+          updateResponse = await this.triggerPostSaveSyncTriggers(oid, currentRec, recordType, 'onTransitionWorkflow', user, updateResponse);
+          
+          this.triggerPostSaveTriggers(oid, currentRec, recordType, 'onTransitionWorkflow', user);
+        }
+      }
+
+      return updateResponse;
+    }
+
+
+    public setWorkflowStepInMetaMetadata(currentRec: any, nextStep: any) {
```

Comment:
This is a bit nit-picky but can i suggest this method be renamed setWorkflowStepRelatedMetadata?
Its a bit of a mouthful but its more accurate than the current name as we set workflow and previousWorkflow as well as metaMetadata